code/modules/urist/modules/wfc/modules/deepmaint/admins.dm

```diff
@@ -0,0 +1,42 @@
+// Adminbuse stuff for deepmaint
+
+/datum/admins/proc/generate_deepmaint()
+	set category = "Fun"
+	set desc = "(Re-)Generate DeepmaintRooms; place the map template first!"
+	set name = "Generate DeepmaintRooms"
+
+	if(!GLOB.wfc_deepmaint_zlevels_by_instance)
+		log_and_message_admins("failed to generate Deepmaint - no Deepmaint Z-levels registered!")
+		return
+
+	for(var/area/map_template/deepmaint_wfc/deepmaint_area in GLOB.wfc_deepmaint_zlevels_by_instance)
+		var/success = deepmaint_area.generate_deepmaint("deepmaint.json", "genmap.json")
+		if(success)
```

Comment:
`success` var is never read. Could we not save an allocation?
```suggestion
		if(deepmaint_area.generate_deepmaint("deepmaint.json", "genmap.json"))
```

Reply:
It's a stack value, it shouldn't matter - whether you assign it to a var or not, it's _there_ in the memory already (otherwise you couldn't if on it).